app/services/langfuse/tasks/score_monitor/score.py

```
@@ -909,21 +920,87 @@ async def check_and_alert(self) -> None:
             f"{len(failing_scores_by_evaluator)} evaluators, sending Slack alerts..."
         )
 
+        # Get action configs for processing before alerts
+        action_configs = await EVALUATOR_ACTIONS()
+        executor = ActionExecutor() if action_configs else None
+
+        # Pre-fetch leads for all call_sids to avoid N+1 queries
+        call_sid_to_lead: Dict[str, Any] = {}
+        if executor and action_configs:
+            for trace_id, call_sid in trace_to_call_sid.items():
```

Ruff (0.15.2)

[warning] Loop control variable `trace_id` not used within loop body

(B007)